src/ucp/core/ucp_mm.c

```diff
@@ -850,52 +921,61 @@ static ucs_status_t ucp_advice2uct(unsigned ucp_advice, uct_mem_advice_t *uct_ad
     return UCS_ERR_INVALID_PARAM;
 }
 
+static ucs_status_t ucp_memh_uct_mem_advise(ucp_context_h context,
+                                            ucp_mem_h memh,
+                                            ucp_md_index_t md_index,
+                                            ucp_md_index_t uct_memh_offset,
+                                            void *address, size_t length,
+                                            uct_mem_advice_t uct_advice)
+{
+    uct_mem_h uct_memh = memh->uct[uct_memh_offset + md_index];
+
+    if (!(context->tl_mds[md_index].attr.flags & UCT_MD_FLAG_ADVISE) ||
+        (uct_memh == NULL)) {
+        return UCS_OK;
+    }
+
+    return uct_md_mem_advise(context->tl_mds[md_index].md, uct_memh, address,
+                             length, uct_advice);
+
+}
+
 ucs_status_t
 ucp_mem_advise(ucp_context_h context, ucp_mem_h memh,
                ucp_mem_advise_params_t *params)
 {
     ucs_status_t status, tmp_status;
-    int md_index;
+    ucp_md_index_t md_index;
     uct_mem_advice_t uct_advice;
-    uct_mem_h uct_memh;
 
     if (!ucs_test_all_flags(params->field_mask,
                             UCP_MEM_ADVISE_PARAM_FIELD_ADDRESS|
                             UCP_MEM_ADVISE_PARAM_FIELD_LENGTH|
                             UCP_MEM_ADVISE_PARAM_FIELD_ADVICE)) {
         return UCS_ERR_INVALID_PARAM;
     }
-
```

Comment:
no need to remove here and below in this func

Reply:
oh, it seems it was added mistakenly in VS code
fixed